auth/auth.go

```diff
@@ -145,6 +156,7 @@ func tradeIDBindToken(ctx context.Context, k8sToken, idPool, idProvider string)
 	req.Header.Set("Content-Type", "application/json")
 	client := &http.Client{}
 	resp, err := client.Do(req)
+	resp, err := http.DefaultClient.Do(req)
```

Comment:
Is there a benefit to the default client? I'm hesitant because its a global that could get modified by dependencies, thought looking at this now the http request should be built with the context which is missing...